lib/jsonapi/serializer.ex

```diff
@@ -22,12 +22,10 @@ defmodule JSONAPI.Serializer do
     {to_include, encoded_data} = encode_data(view, data, conn, query_includes)
 
     %{
-      links: %{
-        self: view.url_for(data, conn)
-      },
       data: encoded_data,
       included: flatten_included(to_include)
     }
+    |> merge_links(data, view, conn, Application.get_env(:jsonapi, :remove_links, false))
```

Comment:
Could we refactor this to avoid piping into a single function?

Reply:
fixed!